add clarity around methods that can be used to set meta-data (#1916)
buildkite/docs: pages/pipelines/build_meta_data.md

## Setting data

> You can set meta-data using the command line or in a script. Both of these methods use the `buildkite-agent` cli with the `meta-data` command. Meta-data can only be set using the `buildkite-agent meta-data set` command, setting meta-data via API or other methods is not possible.

Reviewer: Hmmm how about simplifying this to:

```suggestion
The agent's `meta-data` command is the only method for setting meta-data. You can run the command from the command line or in a script.
```

Or, if there should be more explanation, then:

```suggestion
You can set meta-data using the command line or in a script. Both methods use the agent's `meta-data` command because setting meta-data must happen in the agent context.
```

What do you think? Is it enough to call this out as the only method, or do we need to explicitly mention where it's not possible?
